routes/users.js

```diff
@@ -111,6 +111,11 @@ router.post(
   usersHandler.blockAnonUserFromChat
 );
 
-router.get('/check-allow-dm/:post_id', auth.isAuth, usersHandler.checkAllowAnonDM);
+router.get(
+  '/check-allow-dm',
+  auth.isAuth,
+  validate(UserValidation.checkAllowDm),
+  usersHandler.checkAllowAnonDM
+);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new route definition for '/check-allow-dm' is missing the ':post_id' parameter in the URL. This could lead to issues if the handler function `checkAllowAnonDM` expects a post_id as a route parameter. However, if the post_id is now being passed as a query parameter or in the request body, this change would be fine.

If the `post_id` is still expected as a route parameter, the fix would be:

```diff
-  '/check-allow-dm',
+  '/check-allow-dm/:post_id',
```

Also, it's good to see that validation middleware (`validate(UserValidation.checkAllowDm)`) has been added to the route. This will help ensure that the incoming requests are properly validated before they reach the handler function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->